Add sensors for AsusWRT using http(s) library

ollo69 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Add CPU, Memory and Uptime sensors provided by AsusWrt Http(s) library

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/34511

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

ollo69 committed:
Additional sensors for AsusWRT using http(s) library

PeteRager reviewed (commented):
We should add to the overall description that you have also refactored the temperature sensors into a list format.  If that's an independent change, it could be done in a separate PR

ollo69 committed:
Remove temperature sensors refactor from PR

ollo69 commented:
> We should add to the overall description that you have also refactored the temperature sensors into a list format. If that's an independent change, it could be done in a separate PR

Yes, ok, I did just because I implemented this for CPU sensors. I removed from this PR and I will create a new one with this change after that this will be merged.

ollo69 committed:
Fix test function name

PeteRager commented on homeassistant/components/asuswrt/strings.json:
```diff
@@ -88,6 +88,48 @@
         "name": "6GHz Temperature"
+      },
+      "cpu_usage": {
+        "name": "Cpu usage"
```
Since CPU is an acronym for central processing unit we should upper case it.

ollo69 replied:
Changed as requested

PeteRager commented on homeassistant/components/asuswrt/strings.json:
```diff
@@ -88,6 +88,48 @@
+        "name": "Cpu usage"
+      },
+      "cpu_core1_usage": {
+        "name": "Cpu core1 usage"
```
The should be "CPU core 1 usage", etc.

ollo69 replied:
Changed as requested

PeteRager reviewed (changes requested):
Looks good, just a couple of suggestions on the string names.

ollo69 committed:
Change translation a suggested

joostlek commented on homeassistant/components/asuswrt/sensor.py:
```diff
@@ -46,6 +63,18 @@ class AsusWrtSensorEntityDescription(SensorEntityDescription):
+CPU_SENSORS: tuple[AsusWrtSensorEntityDescription, ...] = tuple(
+    AsusWrtSensorEntityDescription(
+        key=sens_key,
+        translation_key=transl_key,
```
We can use translation_placeholders here.

ollo69 replied:
Done

joostlek commented on tests/components/asuswrt/test_sensor.py:
```diff
@@ -309,6 +319,88 @@ async def test_temperature_sensors_http(hass: HomeAssistant, connect_http) -> No
+    # initial devices setup
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+    async_fire_time_changed(hass, utcnow() + timedelta(seconds=30))
+    await hass.async_block_till_done()
```
idem

ollo69 commented on tests/components/asuswrt/test_sensor.py:
```diff
@@ -309,6 +319,88 @@ async def test_temperature_sensors_http(hass: HomeAssistant, connect_http) -> No
+    # initial devices setup
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+    async_fire_time_changed(hass, utcnow() + timedelta(seconds=30))
+    await hass.async_block_till_done()
```
Could you provide an example?

joostlek replied:
```py
freezer.tick(timedelta(seconds=30))
async_fire_time_changed(hass)
await hass.async_block_till_done()
```

ollo69 replied:
Ok. I replaced everywhere

ollo69 committed:
Requested changes
